### ISSUE TITLE
Training procedure and loss weight

### ISSUE THREAD

**Issue by abrosua** [shengzesnail/PIV-LiteFlowNet-en#1]
I have 2 questions/issues regarding the training procedure of the PIV-LiteFlowNet-en and the usage of the loss weights value for the training process,

**Issue 1**
In the original LiteFlowNet paper ([Hui et al., 2018](https://arxiv.org/pdf/1805.07036.pdf)), it's written that they trained the network stage-wise (e.g., Train the NetC and M_6:S_6 first, then the R_6, and the following level 5 to 2).
How about the PIV-LiteFlowNet-en? Does it follow the same training procedure or simultaneously train all the network stage?

**Issue 2**
It's written in your paper that you used a specific value for the weighting coefficients:
> Based on a number of trials, λ_0 is set as 0.01 while the others are determined to be 0.001.

However, the loss weights that are being used in the [training prototext](https://github.com/shengzesnail/PIV-LiteFlowNet-en/blob/128768b53a52c245883b4048cdd57698617ca4cf/caffe/models/training_template/PIV-LiteFlowNet-en_train.prototxt.template#L818) are the ones that similar to the original LiteFlowNet, with the addition for the extension Level 1 (e.g., 0.32, 0.08, 0.02, 0.01, 0.005, and 0.01).
So, which weight coefficients are being used to train the PIV-LiteFlowNet-en?

Thank you very much for your attention, I'm looking forward for your answers.

**Comment by shengzesnail** [comment]
Sorry for replying late.
Issue 1: PIV-LiteFlowNet-en is trained simultaneously for the whole network.
Issue 2: The weight coefficients used for final PIV-LiteFlowNet-en are the ones stated in the paper (λ_0 is set as 0.01 while the others are determined to be 0.001). For now, I think the training prototxt you mentioned is the one used for 'models/PIV-LiteFlowNet-en/PIV-LiteFlowNet-en_withOriginalLoss.caffemodel'. I will check the files. Thanks.

**Comment by abrosua** [comment]
Thank you very much for the replies

**Comment by abrosua** [comment]
I'm sorry to reopen this issue again, I have another questions regarding the training procedure.

1. If I'm not mistaken, you didn't explicitly state the proportion of the training and test set. Did you use a specific fraction for the dataset (e.g., 80% train and 20% test) or any other method?

2. And for the learning rate, did you define a specific value or following the similar learning rate procedure in the original [LiteFlowNet](https://arxiv.org/pdf/1805.07036.pdf)?

Regards

**Comment by shengzesnail** [comment]
Hi abrosua,

1. For the dataset please refer to another [(repo)](https://github.com/shengzesnail/PIV_dataset.git). Training data and testing data are given in the files FlowData_train.list and FlowData_test.list, respectively. In fact, only a small amount of data (less than 20%) is used for testing. 
2. I follow the strategy of FlowNet, starting with a value like 1e-4 and then deviding it by 2 every 100k iterations. For the presented model, it requires about 1000k iterations. 
Thanks for your attention.

**Comment by abrosua** [comment]
Thanks for the answers, I'm closing this issue now.
